src/Raven.Client/Http/RequestExecutor.cs

```diff
@@ -496,6 +496,8 @@ public virtual async Task<bool> UpdateTopologyAsync(UpdateTopologyParameters par
                     {
                         _nodeSelector = new NodeSelector(topology);
 
+                        ForTestingPurposes?.WaitForRecoredfastest.Invoke(_nodeSelector);
```

Comment:
Shouldn't this be `WaitForRecoredfastest?.Invoke(_nodeSelector);`

Reply:
Maybe it is better to put it inside the `if` block and call it `OnBeforeScheduleSpeedTest` ?

Reply:
Done